openedx/course-discovery
# feat: add contentful reqs and util

[PROD-3023](https://2u-internal.atlassian.net/browse/PROD-3023)
Adds Contentful requirements and utility function to read data from there.

### Testing Instructions
1. Run python shell in discovery shell: `make discovery-shell` `./manage.py shell`
2. Set the following variables in `private.py` file:
```
CONTENTFUL_SPACE_ID = <space_id>
CONTENTFUL_CONTENT_DELIVERY_API_KEY = <api_key>
```
3. Run the following code in shell to read the data from contentful:
```
from course_discovery.apps.course_metadata.utils import get_data_from_contentful
get_data_from_contentful('degreeDetailPage')
```
Expect to see a response like `<Entry[degreeDetailPage] id='3BjaNC0mKiqCtsuaLKCSPC'>`
